frame/beefy/src/lib.rs

```diff
@@ -98,22 +98,27 @@ pub mod pallet {
 impl<T: Config> Pallet<T> {
 	/// Return the current active BEEFY validator set.
 	pub fn validator_set() -> ValidatorSet<T::BeefyId> {
-		ValidatorSet::<T::BeefyId> { validators: Self::authorities(), id: Self::validator_set_id() }
+		let validators: Vec<T::BeefyId> = Self::authorities();
+		let id: beefy_primitives::ValidatorSetId = Self::validator_set_id();
+		// Safe to unwrap() since `Self::authorities()` guaranteed to be non-empty
+		// by Self::initialize_authorities().
+		ValidatorSet::<T::BeefyId>::new(validators, id).unwrap()
```

Comment:
We never use `unwrap`. Please only use `expect()` with a proof why it cannot fail, however since that's part of the runtime code it's almost always much better to fallback to some default (even if it's dummy; or change the return type to `Option`). The reason for not-panicking in runtime is that if this code gets triggered accidentally the chain is doomed forever which is almost impossible to recover from - having some data might just lead to a locally inconsistent/invalid state, but that's better (and easier to recover from) than a complete stall.